File: Docker/Test03/StudentApp.java
import java.util.ArrayList;
import java.util.Scanner;

public class StudentApp {

    static class Student {
        int id;
        String name;
        String course;
        double marks;

        Student(int id, String name, String course, double marks) {
            this.id = id;
            this.name = name;
            this.course = course;
            this.marks = marks;
        }

        void display() {
            System.out.println(
                "ID: " + id +
                " | Name: " + name +
                " | Course: " + course +
                " | Marks: " + marks
            );
        }
    }

    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);
        ArrayList<Student> students = new ArrayList<>();

        students.add(new Student(101, "Kalhara", "Computer Science", 85.5));
        students.add(new Student(102, "Kasun", "Software Engineering", 78.0));
        students.add(new Student(103, "Nimal", "Information Technology", 91.0));

        while (true) {

            System.out.println("\n===== Student Management System =====");
            System.out.println("1. View Students");
            System.out.println("2. Add Student");
            System.out.println("3. Search Student");
            System.out.println("4. Exit");
            System.out.print("Enter your choice: ");

            int choice = scanner.nextInt();
            scanner.nextLine();

            switch (choice) {

                case 1:
                    System.out.println("\n--- Student List ---");

                    for (Student student : students) {
                        student.display();
                    }
                    break;

                case 2:
                    System.out.print("Enter Student ID: ");
                    int id = scanner.nextInt();
                    scanner.nextLine();

                    System.out.print("Enter Student Name: ");
                    String name = scanner.nextLine();

                    System.out.print("Enter Course: ");
                    String course = scanner.nextLine();

                    System.out.print("Enter Marks: ");
                    double marks = scanner.nextDouble();

                    students.add(
                        new Student(id, name, course, marks)
                    );

                    System.out.println("Student added successfully!");
                    break;

                case 3:
                    System.out.print("Enter Student ID to search: ");
                    int searchId = scanner.nextInt();

                    boolean found = false;

                    for (Student student : students) {
                        if (student.id == searchId) {
                            student.display();
                            found = true;
                            break;
                        }
                    }

                    if (!found) {
                        System.out.println("Student not found!");
                    }

                    break;

                case 4:
                    System.out.println("Application closed.");
                    scanner.close();
                    return;

                default:
                    System.out.println("Invalid choice!");
            }
        }
    }
}

File: LB05_B/Test04/student_app.py
class Student:

    def __init__(self, student_id, name, course, marks):
        self.student_id = student_id
        self.name = name
        self.course = course
        self.marks = marks

    def display(self):
        print(
            f"ID: {self.student_id} | "
            f"Name: {self.name} | "
            f"Course: {self.course} | "
            f"Marks: {self.marks}"
        )


students = [
    Student(101, "Kalhara", "Computer Science", 85.5),
    Student(102, "Kasun", "Software Engineering", 78.0),
    Student(103, "Nimal", "Information Technology", 91.0)
]


def view_students():
    print("\n--- Student List ---")

    for student in students:
        student.display()


def add_student():
    print("\n--- Add Student ---")

    student_id = int(input("Enter Student ID: "))
    name = input("Enter Student Name: ")
    course = input("Enter Course: ")
    marks = float(input("Enter Marks: "))

    student = Student(student_id, name, course, marks)
    students.append(student)

    print("Student added successfully!")


def search_student():
    print("\n--- Search Student ---")

    student_id = int(input("Enter Student ID: "))

    for student in students:

        if student.student_id == student_id:
            print("\nStudent Found!")
            student.display()
            return

    print("Student not found!")


def main():

    while True:

        print("\n==============================")
        print("   STUDENT MANAGEMENT SYSTEM")
        print("==============================")
        print("1. View Students")
        print("2. Add Student")
        print("3. Search Student")
        print("4. Exit")
        print("==============================")

        choice = input("Enter your choice: ")

        if choice == "1":
            view_students()

        elif choice == "2":
            add_student()

        elif choice == "3":
            search_student()

        elif choice == "4":
            print("Application closed.")
            break

        else:
            print("Invalid choice!")


if __name__ == "__main__":
    main()

